src/heretic/g_game.c

```diff
@@ -1403,10 +1404,24 @@ void G_PlayerFinishLevel(int player)
 
     if (!deathmatch)
     {
+        current_artifact = p->readyArtifact; // [crispy]
+
         for (i = 0; i < 16; i++)
         {
             P_PlayerUseArtifact(p, arti_fly);
         }
+
+        // [crispy] restore active artifact after removing wings
+        for (i = 0; i < p->inventorySlotNum; i++)
+        {
+            if (p->inventory[i].type == current_artifact)
+            {
+                p->readyArtifact = current_artifact;
+                curpos = inv_ptr = i;
+                curpos = (curpos > 6) ? 6 : curpos;
```

Comment:
Why is this line necessary? (same question below)

Reply:
There are 10 different artifacts in the game, but Heretic can only display a maximum of 7 at a time. When you have more than 7, there is a simple scrolling system in place to set which 7 are in view. `curpos` is the position of the item frame on the screen, and clamping to 6 keeps the frame from being drawn out of bounds.

That clamp to 6 is present elsewhere in `g_game.c` and `p_user.c`. Perhaps I should un-magic the 6 (macro) to make the code more clear?